provider/aws/aws.go

```diff
@@ -324,69 +323,66 @@ func (p *AWSProvider) zones(ctx context.Context) (map[string]*profiledZone, erro
 	log.Debug("Refreshing zones list cache")
 
 	zones := make(map[string]*profiledZone)
-	var profile string
-	var tagErr error
-	f := func(resp *route53.ListHostedZonesOutput, lastPage bool) (shouldContinue bool) {
-		for _, zone := range resp.HostedZones {
-			if !p.zoneIDFilter.Match(aws.StringValue(zone.Id)) {
-				continue
-			}
 
-			if !p.zoneTypeFilter.Match(zone) {
-				continue
-			}
+	for profile, client := range p.clients {
+		paginator := route53.NewListHostedZonesPaginator(client, &route53.ListHostedZonesInput{})
+		for paginator.HasMorePages() {
+			var tagErr error
 
-			if !p.domainFilter.Match(aws.StringValue(zone.Name)) {
-				if !p.zoneMatchParent {
+			resp, err := paginator.NextPage(ctx)
+			if err != nil {
+				var te *route53types.ThrottlingException
+				if errors.As(err, &te) {
+					log.Warnf("Skipping AWS profile %q due to provider side throttling: %v", profile, te.ErrorMessage())
 					continue
 				}
-				if !p.domainFilter.MatchParent(aws.StringValue(zone.Name)) {
+				// nothing to do here. Falling through to general error handling
+				return nil, provider.NewSoftError(fmt.Errorf("failed to list hosted zones: %w", err))
```

Comment:
So if we have a permission error we try to survive?

Reply:
Agreed, though I kep the same existing logic for this: https://github.com/kubernetes-sigs/external-dns/pull/4640/files#diff-95fde44621b48d924a9ad43138223f86d3012e63c65400f7051b7cad08096052L381